src/factory.rs

```diff
@@ -1,37 +1,82 @@
 use crate::*;
+use std::marker::PhantomData;
+use std::sync::atomic::{AtomicPtr, Ordering};
+
+/// Attempts to load and cache the factory interface for the given WinRT class.
+pub struct FactoryCache<C: RuntimeName, I: ComInterface> {
+    pub shared: AtomicPtr<std::ffi::c_void>,
+    pub _c: PhantomData<C>,
+    pub _i: PhantomData<I>,
```

Comment:
Unfortunately, all of these members must be public because Rust doesn't support const functions on generics and the callsite has to initialize these fields because there's now `new` method to wrap it all up.